specs/online_order_spec.rb

```diff
@@ -3,84 +3,132 @@
+      online_order = Grocery::OnlineOrder.new(1,products,3)
+      online_order.order_status.must_equal :pending
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
       # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      online_order = Grocery::OnlineOrder.new(1337, products, 26, :complete)
+
+      sum = products.values.inject(0, :+)
+      expected_total = sum + (sum * 0.075).round(2) + 10
+
+      online_order.total.must_equal expected_total
     end
 
     it "Doesn't add a shipping fee if there are no products" do
       # TODO: Your test code here!
+      products = {}
+      online_order = Grocery::OnlineOrder.new(1337, products, 26, :complete)
+      expected_total = 0
+      online_order.total.must_equal expected_total
     end
   end
 
-  describe "#add_product" do
-    it "Does not permit action for processing, shipped or completed statuses" do
+  xdescribe "#add_product" do
+    xit "Does not permit action for processing, shipped or completed statuses" do
       # TODO: Your test code here!
+      online_order_shipped = Grocery::OnlineOrder.new(27, {"Apples" => 61.87, "Garlic" => 64.36}, 27, :shipped)
+      online_order_shipped.add_product("apple", 12.3).must_be_nil
+
+      online_order_processing = Grocery::OnlineOrder.new(16, {"Ricemilk" => 5.07}, 10, :processing)
+      online_order_processing.add_product("apple", 12.3).must_be_nil
+
+      online_order_complete = Grocery::OnlineOrder.new(38, {"Kale" => 90.99, "Miso" => 90.2}, 2, :complete)
+      online_order_complete.add_product("apple", 12.3).must_be_nil
     end
 
+
     it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+      online_order_pending = Grocery::OnlineOrder.new(6, {"Peaches" => 46.34}, 14, :pending)
+      online_order_pending.must_respond_to(:add_product)
+      online_order_pending.add_product("apple", 12.3).must_equal true
+      online_order_paid = Grocery::OnlineOrder.new(13, {"Lettuce" => 37.94, "Paper" => 23.93, "Flaxseed" => 23.67}, 19, :paid)
+      online_order_paid.must_respond_to(:add_product)
+      online_order_paid.add_product("apple", 12.3).must_equal true
     end
   end
 
+
   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
-      # TODO: Your test code here!
+      # TODO: Your test code here
+      Grocery::OnlineOrder.all.must_be_kind_of Array
+      Grocery::OnlineOrder.all[0].must_be_kind_of Grocery::OnlineOrder
     end
 
     it "Returns accurate information about the first online order" do
       # TODO: Your test code here!
+      product_list = {"Lobster"=> 17.18, "Annatto seed"=> 58.38, "Camomile"=> 83.21}
+      Grocery::OnlineOrder.all.first.id.must_equal 1
+      Grocery::OnlineOrder.all.first.products.must_equal product_list
+      Grocery::OnlineOrder.all.first.customer_id.must_equal 25
+      Grocery::OnlineOrder.all.first.order_status.must_equal :complete
```

Comment:
Every time you have `Grocery::OnlineOrder.all.first`, you're calling and invoking and running `Grocery::OnlineOrder.all`. Which isn't a bad thing!

But it would be all the same/have less code if you just assigned the result of `Grocery::OnlineOrder.all` to a local variable at the top of this method:

```
order = Grocery::OnlineOrder.all.first
order.id.must_equal 1
order.products.must_equal product_list
order.customer_id.must_equal 25
order.status.must_equal :complete
```